conda/core/link.py

```diff
@@ -238,7 +238,7 @@ def execute(self):
         assert not context.dry_run
 
         try:
-            self._execute(self.prefix_action_groups)
+            self._execute(tuple(concat(interleave(itervalues(self.prefix_action_groups)))))
```

Comment:
I often find it better for debugging purposes to have data prep steps separate from function calls.  I find that it makes tweaking and inspecting what goes into the function easier.  This is not blocking this PR - just something to think about.